### PR TITLE
add Trigger and/or python API

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?

add and/or python API

## How was this patch tested?

unit tests

## Related links or issues (optional)

#2672

### CODE REVIEW

File: pyspark/bigdl/optim/optimizer.py [hunk 1]
```diff
@@ -253,6 +253,45 @@ def __init__(self, min, bigdl_type="float"):
         """
         JavaValue.__init__(self, None, bigdl_type, min)
 
+class And(JavaValue):
```
Comment (zhangxiaoli73) [inline pyspark/bigdl/optim/optimizer.py]: Change another name, TriggerAnd?

File: pyspark/bigdl/optim/optimizer.py [hunk 2]
```diff
@@ -253,6 +253,45 @@ def __init__(self, min, bigdl_type="float"):
...
+        """
+        Create a And trigger.
+
+
+        :param first: first Trigger
+        :param other: other Trigger
+        """
+        JavaValue.__init__(self, None, "float", first, list(other))
+
+class Or(JavaValue):
```
Comment (zhangxiaoli73) [inline pyspark/bigdl/optim/optimizer.py]: Same

File: docs/docs/APIGuide/Triggers.md [hunk 3]
```diff
@@ -95,3 +95,26 @@ s reached.
...
+## Or
+**Scala:**
+```scala
+  val trigger = Trigger.and(Trigger.minLoss(0.01), Trigger.maxEpoch(2))
+```
+**Python:**
+```python
+  trigger = Or(MinLoss(0.01), MaxEpoch(2))
+``` 
+A trigger contains other triggers and triggers when all of them trigger (logical OR)
```
Comment (zhangxiaoli73) [inline docs/docs/APIGuide/Triggers.md]: Comments not clear. Actually, user can use this api to do early stop.

Comment (qiuxin2012) [reply docs/docs/APIGuide/Triggers.md]: As early stop is just a specific use case of Or, so I add an example to explain this usage.

**Comment by qiuxin2012** [comment]
jenkins passed

**Pull request merged**